src/main/java/ogc/rs/apiserver/handlers/StacItemByIdAuthZHandler.java

```diff
@@ -0,0 +1,117 @@
+package ogc.rs.apiserver.handlers;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.ext.web.RoutingContext;
+import ogc.rs.apiserver.util.AuthInfo;
+import ogc.rs.apiserver.util.OgcException;
+import ogc.rs.database.DatabaseService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import java.util.UUID;
+import static ogc.rs.apiserver.handlers.DxTokenAuthenticationHandler.USER_KEY;
+import static ogc.rs.common.Constants.DATABASE_SERVICE_ADDRESS;
+import static ogc.rs.common.Constants.UUID_REGEX;
+
+public class StacItemByIdAuthZHandler implements Handler<RoutingContext> {
+
+    Vertx vertx;
+
+    private final DatabaseService databaseService;
+
+    private static final Logger LOGGER = LogManager.getLogger(StacItemByIdAuthZHandler.class);
+
+    public StacItemByIdAuthZHandler(Vertx vertx) {
+        this.vertx = vertx;
+        this.databaseService = DatabaseService.createProxy(vertx, DATABASE_SERVICE_ADDRESS);
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext){
+
+        if(System.getProperty("disable.auth") != null) {
+            routingContext.put("shouldCreate", false);
```

Comment:
Put `shouldCreate` as a constant of the handler and use that constant to fetch it from the `RoutingContext` in the `ApiServerVerticle` (e.g. `DxTokenAuthenticationHandler.USER_KEY`)